### PR TITLE
TELCODOCS-1682 - RN - Telco RAN 4.16 - MetalLB - FRRK8s integration

### PR DESCRIPTION
**Pull request by slovern**
Release Note for [TELCODOCS-1736](https://issues.redhat.com//browse/TELCODOCS-1736) - Telco RAN 4.16 - MetalLB - FRRK8s integration - [TELCODOCS-1682](https://issues.redhat.com//browse/TELCODOCS-1682) 

Version(s): 4.16

Issue:
RN for Telco RAN 4.16 - https://issues.redhat.com/browse/TELCODOCS-1736
Integration of MetalLB and FRR-K8s - https://issues.redhat.com/browse/TELCODOCS-1682

Link to docs preview: [Networking](https://77411--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-16-release-notes.html#ocp-4-16-networking_release-notes)
- Integration of MetalLB and FRR-K8s (Technology Preview)

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: release_notes/ocp-4-16-release-notes.adoc [hunk 1]
```diff
@@ -1004,6 +1012,11 @@ In the following tables, features are marked with the following statuses:
 |General Availability
 |General Availability
 
+|Integration of MetalLB and FRR-K8s
+|Not available
```
Comment (jneczypor) [inline release_notes/ocp-4-16-release-notes.adoc:1016]:
```suggestion
|Not Available
```

File: release_notes/ocp-4-16-release-notes.adoc [hunk 2]
```diff
@@ -1004,6 +1012,11 @@ In the following tables, features are marked with the following statuses:
...
+|Not available
```
Comment (jneczypor) [inline release_notes/ocp-4-16-release-notes.adoc:1017]:
```suggestion
|Not Available
```

**Review by jneczypor — COMMENTED** [review]
Just a couple of quick changes. I'm new to peer review, so @michaelryanpeter will follow with his final review. Thanks for your patience!

File: release_notes/ocp-4-16-release-notes.adoc [hunk 3]
```diff
@@ -208,6 +208,14 @@ If you have already upgraded your cluster to {product-title} {product-version} a
 * An IPv6 network for API and Ingress services on the second `machineNetwork` configuration.
 
 For more information, see xref:../networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html#nw-dual-stack-convert_converting-to-dual-stack[Converting to a dual-stack cluster network] in _Converting to IPv4/IPv6 dual-stack networking_.
+[id="ocp-4-16-networking-metallb-frr-k8s_{context}"]
+==== Integration of MetalLB and FRR-K8s (Technology Preview)
+
+This release introduces `FRR-K8s`, a Kubernetes based DaemonSet that exposes a subset of the `FRR` API in a Kubernetes-compliant manner.
+As a cluster administrator, you can use the `FRRConfiguration` custom resource (CR) to configure `MetalLB` to use `FRR-K8s` as the backend.
+You can use this to avail of FRR services, for example, receiving routes.
```
[the remark this answers is not in the document]
Comment (slovern) [reply release_notes/ocp-4-16-release-notes.adoc:216]: Thanks for the review @jneczypor  the idea I was trying to get across that FRR-K8s gives MetalLB access to FRR services that were not previously available.
I think "operate" will work here.

**Comment by slovern** [comment]
Thanks again, @jneczypor 
I've updated with your suggestions
@michaelryanpeter Do I need to wait for you to finalise the peer-review stage or can I label this for merge-review?

File: release_notes/ocp-4-16-release-notes.adoc [hunk 4]
```diff
@@ -208,6 +208,14 @@ If you have already upgraded your cluster to {product-title} {product-version} a
 * An IPv6 network for API and Ingress services on the second `machineNetwork` configuration.
 
 For more information, see xref:../networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html#nw-dual-stack-convert_converting-to-dual-stack[Converting to a dual-stack cluster network] in _Converting to IPv4/IPv6 dual-stack networking_.
+[id="ocp-4-16-networking-metallb-frr-k8s_{context}"]
+==== Integration of MetalLB and FRR-K8s (Technology Preview)
+
+This release introduces `FRR-K8s`, a Kubernetes based DaemonSet that exposes a subset of the `FRR` API in a Kubernetes-compliant manner.
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-16-release-notes.adoc:214]: See the repo guidelines for guidance on [general](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references) vs [object](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references) formatting for APIs. _DaemonSet_ should be formatted as either "daemon set" or `DaemonSet`.

File: release_notes/ocp-4-16-release-notes.adoc [hunk 5]
```diff
@@ -208,6 +208,14 @@ If you have already upgraded your cluster to {product-title} {product-version} a
...
+As a cluster administrator, you can use the `FRRConfiguration` custom resource (CR) to configure `MetalLB` to use `FRR-K8s` as the backend.
+You can use this to operate FRR services, such as receiving routes.
+
+For more information, see xref:../networking/metallb/metallb-frr-k8s.adoc[Configuring the integration of MetalLB and FRR-K8s].
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-16-release-notes.adoc:218]: This[ xref needs an anchor id](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references). For detailed examples on how to format xrefs, see the [Cross-referencing by example](https://docs.google.com/document/d/1pavrM-4u8MEmuZFNF3wBeDmcmZi-kAZNRxp1Cze2CkQ/edit).

File: release_notes/ocp-4-16-release-notes.adoc [hunk 6]
```diff
@@ -208,6 +208,14 @@ If you have already upgraded your cluster to {product-title} {product-version} a
 * An IPv6 network for API and Ingress services on the second `machineNetwork` configuration.
 
 For more information, see xref:../networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html#nw-dual-stack-convert_converting-to-dual-stack[Converting to a dual-stack cluster network] in _Converting to IPv4/IPv6 dual-stack networking_.
+[id="ocp-4-16-networking-metallb-frr-k8s_{context}"]
+==== Integration of MetalLB and FRR-K8s (Technology Preview)
+
+This release introduces `FRR-K8s`, a Kubernetes based DaemonSet that exposes a subset of the `FRR` API in a Kubernetes-compliant manner.
+As a cluster administrator, you can use the `FRRConfiguration` custom resource (CR) to configure `MetalLB` to use `FRR-K8s` as the backend.
```
Comment (michaelryanpeter) [inline release_notes/ocp-4-16-release-notes.adoc:215]: As mentioned in the object references guidance, specific object references need to be matched with their object type, or `kind`. 
```suggestion
As a cluster administrator, you can use the `FRRConfiguration` custom resource (CR) to configure `MetalLB` <object_type> to use `FRR-K8s` <object_type> as the backend.
```

**Review by michaelryanpeter — COMMENTED** [review]
I found a few more style issues. Please let me know if you have any questions.

/label peer-review-done
/remove-label peer-review-in-progress
/remove-label peer-review-needed

**Comment by michaelryanpeter** [comment]
@slovern To clarify, this was just a peer review. Thanks!

File: release_notes/ocp-4-16-release-notes.adoc (hunk 4 above)
Comment (slovern) [reply release_notes/ocp-4-16-release-notes.adoc:214]: Thanks for the review @michaelryanpeter

File: release_notes/ocp-4-16-release-notes.adoc (hunk 6 above)
Comment (slovern) [reply release_notes/ocp-4-16-release-notes.adoc:215]: Thanks. `MetalLB` is an operator and `FRR-K8s` is a daemon set - updated accordingly

Comment (michaelryanpeter) [reply release_notes/ocp-4-16-release-notes.adoc:215]: A friendly reminder that in the docs, "Operator" is always capitalized :)

File: release_notes/ocp-4-16-release-notes.adoc (hunk 5 above)
Comment (slovern) [reply release_notes/ocp-4-16-release-notes.adoc]: Thanks for that. Updated accordingly.

File: release_notes/ocp-4-16-release-notes.adoc [hunk 7]
```diff
@@ -208,6 +208,14 @@ If you have already upgraded your cluster to {product-title} {product-version} a
 * An IPv6 network for API and Ingress services on the second `machineNetwork` configuration.
 
 For more information, see xref:../networking/ovn_kubernetes_network_provider/converting-to-dual-stack.html#nw-dual-stack-convert_converting-to-dual-stack[Converting to a dual-stack cluster network] in _Converting to IPv4/IPv6 dual-stack networking_.
+[id="ocp-4-16-networking-metallb-frr-k8s_{context}"]
+==== Integration of MetalLB and FRR-K8s (Technology Preview)
+
+This release introduces `FRR-K8s`, a Kubernetes based `DaemonSet` that exposes a subset of the `FRR` API in a Kubernetes-compliant manner.
```
Comment (stevsmit) [inline release_notes/ocp-4-16-release-notes.adoc:214]: ours docs are *everywhere* with DaemonSet. I see `DaemonSet`, daemon set, Daemonset, daemonset. I do not think it needs to be in backticks, but I think that properly it is DaemonSet. But I'll let you decide how to move forward with it.

This is one of those terms I think needs added to our glossary.

File: release_notes/ocp-4-16-release-notes.adoc [hunk 8]
```diff
@@ -208,6 +208,14 @@ If you have already upgraded your cluster to {product-title} {product-version} a
...
+As a cluster administrator, you can use the `FRRConfiguration` custom resource (CR) to configure the `MetalLB` Operator to use the `FRR-K8s` daemon set as the backend.
```
Comment (stevsmit) [inline release_notes/ocp-4-16-release-notes.adoc:215]: Pretty sure it's just MetalLB Operator (without backticks), unless that has changed. 

Note how DaemonSet is being used here vs. above. Maybe there's a difference in context here, in that the one above is referring to a `DaemonSet` resource and this one maybe isn't, but I'm not sure. Just thought I'd raise that scenario in the event that that is how you are using them.

```suggestion
As a cluster administrator, you can use the `FRRConfiguration` custom resource (CR) to configure the MetalLB Operator to use the `FRR-K8s` daemon set as the backend.
```

File: release_notes/ocp-4-16-release-notes.adoc (hunk 7 above)
Comment (michaelryanpeter) [reply release_notes/ocp-4-16-release-notes.adoc:214]: While I agree that there should be a glossary update, the doc guidelines do use this object as a specific example:
* https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc?plain=1#L1972
* https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc?plain=1#L2007

Comment (michaelryanpeter) [reply release_notes/ocp-4-16-release-notes.adoc:214]: but not a merge blocker IMO.

Comment (stevsmit) [reply release_notes/ocp-4-16-release-notes.adoc:214]: TIL

Comment (michaelryanpeter) [reply release_notes/ocp-4-16-release-notes.adoc:214]: I only found it by searching the repo for DaemonSet :)

**Pull request merged**